src/backend/EdFi.DataManagementService.Backend.Postgresql/PostgresqlDocumentStoreRepository.cs

```diff
@@ -40,18 +40,24 @@ public async Task<UpsertResult> UpsertDocument(IUpsertRequest upsertRequest)
                 }
             };
 
-            if (await cmd.ExecuteNonQueryAsync() == 1)
+            int resultCount = await cmd.ExecuteNonQueryAsync();
+            if (resultCount == 1)
             {
-                _logger.LogDebug("Insert Success - {TraceId}", upsertRequest.TraceId);
+                _logger.LogDebug("Upsert success - {TraceId}", upsertRequest.TraceId);
                 return new UpsertResult.InsertSuccess(upsertRequest.DocumentUuid);
             }
 
-            _logger.LogError("Unknown Error - {TraceId}", upsertRequest.TraceId);
+            _logger.LogError(
+                "Upsert result count was {ResultCount} for {DocumentUuid}, cause is unknown - {TraceId}",
+                resultCount,
+                upsertRequest.DocumentUuid,
+                upsertRequest.TraceId
+            );
```

Comment:
While looking at this class, I realized that we _do know something_ about the error, and we have an opportunity to be more clear about what happened. Also clarified some messages below this.